### DISCUSSION TITLE
Solax Gen3 (QCells HYB-G1) Sensors

### DISCUSSION THREAD

**Discussion by aqualignum** [wills106/homeassistant-solax-modbus#663] [General]
Hi, I have a QCells HYB-G1 Inverter which I believe is a re-branded Solax Gen3. I am able to install this modbus integration and it connects fine via TCP/502 to the DHCP IP on the hardwired LAN port on the inverter. However it looks as if I am missing some configuration items, have out of date firmware, or something may not be compatible.

In the Integration it detects 1 device and 4 entities:
sensor.solax_inverter_temperature
button.solax_battery_awaken
button.solax_unlock_inverter
button.solax_unlock_inverter_advanced

The sensor.solax_inverter_temperature value is being populated accurately (66 degrees F) but I'm not sure how - it's not in any yaml file. So I imagine it must be coming from within the integration code?

How do I add the other sensor data (battery_charge_upper_soc, battery_charge_max_current, etc)? I've seen some examples of this yaml code in other posts but the modbus addresses referenced look quite specific and not sure if they would match my inverter.

Read as many docs as I could find, still stuck, would appreciate any direction. Thank you!

**Comment by robertfili** [comment]
probably inverter not rcognized or locked
